lib_xua/src/core/mixer/mixer.xc

```diff
@@ -603,28 +647,24 @@ static void mixer1(chanend c_host, chanend c_mix_ctl, chanend ?c_mixer2, chanend
         {
             int sampFreq;
             unsigned command = inct(c_host);
+            cmd_pending_flag = 1; // set pending flag. Command will be forwarded to audihub in the next sample period.
+            pending_cmd[0] = command;
 
             switch(command)
             {
                 case XUA_AUDCTL_SET_SAMPLE_FREQ:
                     sampFreq = inuint(c_host);
                     mixer1_mix2_flag = sampFreq > 96000;
-
-                    /* Inform mixer2 (or audio()) about freq change */
-                    outct(c_audio, command);
-                    outuint(c_audio, sampFreq);
+                    pending_cmd[1] = sampFreq;
                     break;
 
                 case XUA_AUD_SET_AUDIO_START:
                     /* Inform mixer2 (or audio()) about change */
-                    outct(c_audio, command);
-                    outuint(c_audio, inuint(c_host));
-                    outuint(c_audio, inuint(c_host));
+                    pending_cmd[1] = inuint(c_host);
```

Comment:
We should add some comments here 

// DSD Mode
// Sample Resolution 

or self comment with variable names

Reply:
Replaced with a generic command structure instead.